speech/google/cloud/speech/streaming/request.py

```diff
@@ -0,0 +1,183 @@
+                             asterisks, e.g. ``'f***'``. If False or
+                             omitted, profanities won't be filtered out.
+
+    :type speech_context: list
+    :param speech_context: A list of strings (max 50) containing words and
+                           phrases "hints" so that the speech recognition
+                           is more likely to recognize them. This can be
+                           used to improve the accuracy for specific words
+                           and phrases. This can also be used to add new
+                           words to the vocabulary of the recognizer.
+
+    :type single_utterance: boolean
+    :param single_utterance: [Optional] If false or omitted, the recognizer
+                             will perform continuous recognition
+                             (continuing to process audio even if the user
+                             pauses speaking) until the client closes the
+                             output stream (gRPC API) or when the maximum
+                             time limit has been reached. Multiple
+                             SpeechRecognitionResults with the is_final
+                             flag set to true may be returned.
+
+                             If true, the recognizer will detect a single
+                             spoken utterance. When it detects that the
+                             user has paused or stopped speaking, it will
+                             return an END_OF_UTTERANCE event and cease
+                             recognition. It will return no more than one
+                             SpeechRecognitionResult with the is_final flag
+                             set to true.
+
+    :type interim_results: boolean
+    :param interim_results: [Optional] If true, interim results (tentative
+                            hypotheses) may be returned as they become
+                            available (these interim results are indicated
+                            with the is_final=false flag). If false or
+                            omitted, only is_final=true result(s) are
+                            returned.
+    """
+    config_request = _make_streaming_config(
+        sample, language_code=language_code, max_alternatives=max_alternatives,
+        profanity_filter=profanity_filter, speech_context=speech_context,
+        single_utterance=single_utterance, interim_results=interim_results)
+
+    # The config request MUST go first and not contain any audio data.
+    yield config_request
+
+    buff = b''
+    for data in sample.stream:
+        # Optimize the request data size to around 100ms.
+        if len(buff) + len(data) >= sample.chunk_size:
+            yield StreamingRecognizeRequest(audio_content=buff)
+            buff = data
+        else:
+            buff += data
+
+    # Clear final contents of buffer.
+    yield StreamingRecognizeRequest(audio_content=buff)
```

Comment:
I've updated per your suggestion.

FYI, when streaming a non-file(mic input), I found that it would return chunks much smaller than what you ask it for. Which would then cause it to issue many requests that were not the optimal ~100ms size.  But adding this doesn't really make any difference in the requests as long as the buffer is there. Obviously it should have less disk reads, which is good.